modules/apps/redirect/redirect-service/src/main/java/com/liferay/redirect/internal/util/PatternUtil.java

```diff
@@ -45,10 +45,20 @@ public static Map<Pattern, String> parse(String[] patternStrings) {
 
 	private static String _normalize(String patternString) {
 		if (patternString.startsWith(StringPool.CARET)) {
+			if (patternString.startsWith(_CARET_AND_SLASH)) {
+				return StringPool.CARET + patternString.substring(2);
+			}
+
 			return patternString;
 		}
 
+		if (patternString.startsWith(StringPool.SLASH)) {
+			patternString = patternString.substring(1);
+		}
+
 		return StringPool.CARET + patternString;
 	}
```

Comment:
Maybe you find this version easier to understand:

```
	private static String _normalize(String patternString) {
		if (patternString.startsWith(StringPool.CARET)) {
			patternString = patternString.substring(1);
		}

		if (patternString.startsWith(StringPool.SLASH)) {
			patternString = patternString.substring(1);
		}

		return StringPool.CARET + patternString;
	}
```

Reply:
Yes, I agree.